### ISSUE TITLE
highlight plugin breaks additionally CSS-classes for codeblocks

### ISSUE THREAD

**Issue by FlTr** [datenstrom/yellow#78]
Short examples:

---
OK:

````
```{.left}

// here some fancy code
// ...

````

---
OK:

````
```javascript

// here some fancy code
// ...

````

---
Not OK:

````
```javascript {.left}

// here some fancy code
// ...

````

---
Have anyone a solution?

**Comment by markseu** [comment]
Looks like [PHP Markdown Extra](https://michelf.ca/projects/php-markdown/extra/#fenced-code-blocks) doesn't support code blocks with classes both inside and outside the braces. I can add support for the syntax `{javascript .left}` in the next version.

**Comment by markseu** [comment]
Should be working now.

**Comment by FlTr** [comment]
great support!

**Comment by FlTr** [comment]
Sorry, but i think it doesn't work. The CSS-class is added but the highlight-tag is ignored. However now it doesn't break the whole layout.

**Comment by markseu** [comment]
Can you do me a favor and try it out on the [Yellow demo](http://demo.datenstrom.se) server?

**Comment by FlTr** [comment]
Uff, I'm a little bit confused. Yes on demo it seems it works, but all code blocks are surrounded with div-tag, and even there is no highlight-tag, their are highlighted ...

**Comment by FlTr** [comment]
This "auto-wrapping" does not appear on my local machine (latest core and highlight-plugin)

**Comment by markseu** [comment]
Can you check the version of core and plugins?

What happens with the latest version: The highlight plugin wraps code blocks in `div` in the HTML, even if it doesn't recognize the language. If you use a normal code block, one without language or class at all, then it generates ```<pre>...</pre>``` HTML.  Is that what you expected or should it be changed?

**Comment by FlTr** [comment]
Ok, my fault. I thought I updated it but there are more dependencies between core and plugins. With latest versions it works. Thanks.

```bash
php yellow.php version
Yellow 0.5.19
YellowCommandline 0.5.3
YellowHighlight 0.5.4
YellowMarkdown 0.5.5
```

Anyway there should be a simpler update procedure.

**Comment by FlTr** [comment]
I think all codeblocks should be handled the same way. So add the additional classes to the ```<pre>``` or surround all blocks with a ```<div>```

**Comment by FlTr** [comment]
An extra wish:
Short descripton of the css-classes from *highlight.css*

**Comment by markseu** [comment]
Let me see what I can do. However, we are using [Geshi syntax highlighting](https://github.com/GeSHi/geshi-1.0) library in the highlight plugin. So the HTML and CSS for highlighted code blocks is coming from the highlighting library.

**Comment by markseu** [comment]
I think I found something that's more predictable. The highlight plugin is only invoked when there's a language identifier. Then the highlight plugin is responsible for generated HTML and CSS. On the other hand if there's no language identifier, then Markdown generates `<pre>...</pre>` without highlighting.

Here are code blocks with highlighting:

    ```javascript
    alert('Hello world!')
    ```
.

    ```{javascript .left}
    alert('Hello world!') 
    ```

Here are code blocks without highlighting:

    ```
    alert('Hello world!')
    ```
.

    ```{.left}
    alert('Hello world!') 
    ```

Hope this works you.

**Comment by markseu** [comment]
Waiting for bug being fixed in michelf/php-markdown/pull/200

**Comment by markseu** [comment]
There's a new Markdown plugin, syntax highlighting supports optional classes. 

    ```javascript {.left}
    alert('Hello world!')
    ```

This should be working now.
